### ISSUE TITLE
`CREATE USER` statement in Postgres not supported

### ISSUE THREAD

**Issue by kayman-mk** [sqlfluff/sqlfluff#5380]
### Search before asking

- [X] I searched the [issues](https://github.com/sqlfluff/sqlfluff/issues) and found no similar issues.

### What Happened

I linted a SQL script for Postgres DB creating a new user with a password. Linter failed with `Found unparsable section`.

```
CREATE USER 1234 WITH PASSWORD '1234';
```

### Expected Behaviour

As the statement is correct the linter should not throw an error message.

### Observed Behaviour

Linter failed with `Found unparsable section`.

```
[L:  1, P:  1]      |file:
[L:  1, P:  1]      |    statement:
[L:  1, P:  1]      |        unparsable:                                           !! Expected: 'statement'
[L:  1, P:  1]      |            code:                                             'CREATE'
[L:  1, P:  7]      |            whitespace:                                       ' '
[L:  1, P:  8]      |            code:                                             'USER'
[L:  1, P: 12]      |            whitespace:                                       ' '
[L:  1, P: 13]      |            single_quote:                                     "'1234'"
[L:  1, P: 19]      |            whitespace:                                       ' '
[L:  1, P: 20]      |            code:                                             'WITH'
[L:  1, P: 24]      |            whitespace:                                       ' '
[L:  1, P: 25]      |            code:                                             'PASSWORD'
[L:  1, P: 33]      |            whitespace:                                       ' '
[L:  1, P: 34]      |            single_quote:                                     "'1234'"
[L:  1, P: 40]      |    statement_terminator:                                     ';'
[L:  1, P: 41]      |    newline:                                                  '\n'
[L:  2, P:  1]      |    newline:                                                  '\n'
```

### How to reproduce

Create a SQL file with `CREATE USER 1234 WITH PASSWORD '1234';` and lint it with `sqlfluff lint --dialect postgres my.sql`

### Dialect

postgres

### Version

2.1.2

### Configuration

no special configuration

### Are you willing to work on and submit a PR to address the issue?

- [ ] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://github.com/sqlfluff/sqlfluff/blob/main/CODE_OF_CONDUCT.md)

**Comment by kayman-mk** [comment]
Just noticed that 2.1.2 is outdated. Updated to 2.3.4, but behavior is still the same.

**Comment by greg-finley** [comment]
Can you really make users like that though?

```
gregoryfinley@Gregorys-MacBook-Pro git-repos % psql
psql (14.8 (Homebrew))
Type "help" for help.

gregoryfinley=# CREATE USER 1234 WITH PASSWORD '1234';
ERROR:  syntax error at or near "1234"
LINE 1: CREATE USER 1234 WITH PASSWORD '1234';
                    ^
gregoryfinley=# 
```

**Comment by kayman-mk** [comment]
Gosh, just a bad substitution here and a misinterpreted error message. Sorry for bothering. Works perfectly!
